### PR TITLE
Version 1.0.0

### PR DESCRIPTION
**Pull request by Raruto**
**Breaking changes:**

- rewritten code by adding a self-contained class
- removed no longer needed `branch` headers
- no longer support deprecated `GitHub Access Token` header

### CODE REVIEW

**Comment by szepeviktor** [comment]
Thank you @Raruto !

If you get the approval of @afragen I'll press "Merge".

**Comment by szepeviktor** [comment]
... as I've forgotten everything about this plugin and GitHub Updater :)

**Comment by Raruto** [comment]
Ok, in addition to that, do you think we could somehow add the word "icon" to the actual plugin name?
(eg. `github-icon-link`, it's definitely easier to find ...)

File: github-link.php [hunk 1]
```diff
@@ -1,180 +1,168 @@
...
-        || isset( $plugin_state->no_update[ $plugin_file ] )
-    ) {
-        $on_wporg = true;
-    }
+    function init_hooks() {
+        include_once ABSPATH . '/wp-admin/includes/plugin.php';
+
+        $this->extra_plugin_headers = [ 'GitHub Plugin URI', 'GitLab Plugin URI', 'Bitbucket Plugin URI' ];
+
+        add_filter( "extra_plugin_headers", [ &$this, "extra_plugin_headers" ] );
```
Comment (afragen) [inline github-link.php:43]: GitHub Updater moved away from using this filter in favor of using get_plugin_data() to find and return header data. It was much less flaky overall.

**Comment by afragen** [comment]
@szepeviktor I have made a comment, but in general I have no issues and @Raruto really doesn’t need my approval. 

FWIW, I appreciate his insightful suggestions. 😉

File: github-link.php (hunk 1 above)
Comment (szepeviktor) [reply github-link.php:43]: All right, after this modification I'll merge this and tag v1.0.0

**Comment by szepeviktor** [comment]
@afragen I do not use GitHub Updater in my clients' projects as Composer is used to manage updates and thus almost all PHP files.

File: github-link.php (hunk 1 above)
Comment (Raruto) [reply github-link.php:43]: @afragen I try and let you know, however from what I remember, is it still not necessary to add additional headers?

- [get_plugin_data](https://developer.wordpress.org/reference/functions/get_plugin_data/)
- [get_file_data](https://developer.wordpress.org/reference/functions/get_file_data/)

Comment (Raruto) [reply github-link.php:43]: I have tried the following:

```php
$this->extra_plugin_headers = [ 'GitHub Plugin URI', 'GitLab Plugin URI', 'Bitbucket Plugin URI' ];

// add_filter( "extra_plugin_headers", [ &$this, "extra_plugin_headers" ] );
```

But doing so the following hooks:

- [plugin_action_links_{$plugin_file}](https://developer.wordpress.org/reference/hooks/plugin_action_links_plugin_file/)
- [network_admin_plugin_action_links_{$plugin_file}](https://developer.wordpress.org/reference/hooks/network_admin_plugin_action_links_plugin_file/)

cannot provide information about theese additional headers.

> GitHub Updater moved away from using this filter in favor of using get_plugin_data() to find and return header data. It was much less flaky overall.

Actually i don't yet understand how your plugin can read them without adding them to the list of known headers..

Comment (szepeviktor) [reply github-link.php:43]: @afragen I think we **really** need you as I have no clue what is going on.

**Comment by Raruto** [comment]
> @afragen I do not use GitHub Updater in my clients' projects as Composer is used to manage updates and thus almost all PHP files.

@afragen as this is really a very small plugin, couldn't we also get it incorporated into your [local-development](https://github.com/afragen/local-development) repo? 

Both info provided by these two plugins are very useful during development (let's say one completes the other ...)

In this way we could also easily install everything via wordpress.org (without necessarily having to go through your other famous plugin ...)

**Comment by afragen** [comment]
@Raruto if you want to make a PR to local-development that's ok.

**Comment by Raruto** [comment]
> @Raruto if you want to make a PR to local-development that's ok.

Ok, got it @afragen

**Comment by Raruto** [comment]
@szepeviktor now `local-development` supports this feature too: https://github.com/afragen/local-development/pull/7

According to what @afragen says, the solution adopted here in `github-link` works but performs many calls to wordpress servers (to check if the plugin exists on wp.org).

For now, as far as I'm concerned, this pull can be considered currently completed.

**Comment by szepeviktor** [comment]
Thank you.
Closing in favor of https://github.com/afragen/local-development/pull/7

**Pull request closed without merging**